src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -126,16 +129,31 @@
 
     private static final Supplier<Provider[]> DEFAULT_PROVIDER_SUPPLIER = ProviderUtil.aggregate(ELYTRON_PROVIDER_SUPPLIER, Security::getProviders);
 
-    static final String NS_ELYTRON_1_0 = "urn:elytron:1.0";
-    static final String NS_ELYTRON_CLIENT_1_1 = "urn:elytron:client:1.1";
+    static final Map<String, Version> KNOWN_NAMESPACES;
 
-    private static final Set<String> KNOWN_NAMESPACES;
+    private static final class Version {
+        final int major;
+        final int minor;
+
+        Version(int major, int minor) {
+            this.major = major;
+            this.minor = minor;
+        }
+
+        boolean isAtLeast(int major, int minor) {
```

Comment:
I think this works for now but we possibly have more complex scenarios coming later e.g. we may add an element to 2.0.1 AND 1.0.3 so we could not say at least 1.0.3 as that would incorrectly accept an element for 2.0.0.